src/rgw/rgw_rest_swift.cc

```diff
@@ -1325,9 +1325,86 @@ void RGWBulkDelete_ObjStore_SWIFT::send_response()
   rgw_flush_formatter_and_reset(s, s->formatter);
 }
 
+map<string, RGWInfo_ObjStore_SWIFT::list_data> RGWInfo_ObjStore_SWIFT::swift_info =
+  {
+    {"bulk_delete", NULL},
+    {"container_quotas", NULL},
+    {"swift", RGWInfo_ObjStore_SWIFT::list_swift_data},
+    {"tempurl", RGWInfo_ObjStore_SWIFT::list_tempurl_data},
+    {"slo", RGWInfo_ObjStore_SWIFT::list_slo_data},
+    {"account_quotas", NULL},
+    {"staticweb", NULL},
+  };
+
+void RGWInfo_ObjStore_SWIFT::send_response()
+{
+  if (op_ret <  0) {
+    op_ret = STATUS_NO_CONTENT;
+  }
+  set_req_state_err(s, op_ret);
+  dump_errno(s);
+  end_header(s, this);
+
+  s->formatter->open_object_section("info");
+
+  for (auto it = swift_info.begin(); it != swift_info.end(); it++)
+  {
+    if (!it->second)
+    {
+      s->formatter->open_object_section((it->first).c_str());
+      s->formatter->close_section();
+    }
+    else
+    {
+      it->second(*(s->formatter), *(s->cct->_conf));
+    }
+  }
+
+  s->formatter->close_section();
```

Comment:
the parts of an op that do actual work (including writing to the formatter) usually happen in the `execute()` function. can you move this block, starting with the `open_object_section()`?